src/Symfony/Component/Messenger/MessageBus.php

```diff
@@ -54,6 +55,10 @@ public function getIterator(): \Traversable
 
     public function dispatch(object $message, array $stamps = []): Envelope
     {
+        if ($message instanceof SelfStampableInterface) {
```

Comment:
im not sure if people expect `SelfStampableInterface` works for any `MessageBusInterface`

Reply:
I didn't find a better place to automatically add the `stamp` since it's the place where
```
$envelope = Envelope::wrap($message, $stamps);
```
is done.

And then `return $middlewareIterator->current()->handle($envelope, $stack);` is done.

But if I introduce a `SelfStampableMiddleware` to automatically add the stamp, I cannot be sure:
- The middleware is used
- The middleware is the first one

So the same issue would kinda occurs no ?

Reply:
we can wire the middleware in frameworkbundle, but at least then there's infrastructure for others to use the same feature.